File: 3208-alternating-groups-ii/3208-alternating-groups-ii.cpp
class Solution {
public:
    int numberOfAlternatingGroups(vector<int>& colors, int k) {
       int n=colors.size();
       int ans=0,cnt=1; // cnt for prefix sum of alternating colors
for(int i=1;i<n+k-1;i++){ 
if(colors[i%n]!=colors[(i-1)%n]) cnt++;
else cnt=1;
if(cnt>=k) ans++; // alternating colors>k includes our ans
}       
return ans;
    }
};

File: 3208-alternating-groups-ii/3208-alternating-groups-ii.java
class Solution {
    public int numberOfAlternatingGroups(int[] colors, int k) {
        int len = colors.length, cnt = 1, ans = 0;

        for(int idx = 1; idx < (len + k - 1); idx++) {
            if(colors[idx % len] != colors[(idx - 1)%len]){
                cnt++;
                if(cnt == k) {
                    ans++;
                    cnt -= 1;
                }
            } else {
                cnt = 1;
            }        
        }
        return ans;
    }
}